llvm/include/llvm/Support/SimpleTable.h

```diff
@@ -0,0 +1,143 @@
+//==-- SimpleTable.h -- tabular data simple transforms and I/O -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Defines a simple model for a tabular data container with simple operations
+// over rows and columns. Columns are named and referenced by name.
+// Major use case is to model dynamically-sized "2D" sets of output files by
+// tools like post-link and being able to manipulate columns - for example
+// replace a column listing files with bitcode with a column of .spv files.
+//
+// TODO May make sense to make the interface SQL-like in future if evolves.
+// TODO Use YAML as serialization format.
+//
+// Example of a table:
+// [Code|Symbols|Properties]
+// a_0.bc|a_0.sym|a_0.props
+// a_1.bc|a_1.sym|a_1.props
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/MemoryBuffer.h"
+
+#include <list>
+#include <map>
+#include <memory>
+#include <string>
+
+using namespace llvm;
+
+namespace llvm {
+namespace util {
+
+// The string tabular data abstraction.
+class SimpleTable {
```

Comment:
```suggestion
class StringTable {
```
or 
```suggestion
class SimpleStringTable {
```
?

Reply:
It is string table today, but can be extended to cover more types if needed - e.g. cells may have integer number semantics. So I'd leave it as is. Add a comment?

Reply:
added a comment

Reply:
Could you update the comment above the class declaration, please?
It says `// The string tabular data abstraction.`

Reply:
ok